crates/starknet/src/transactions/declare_transaction.rs

```diff
@@ -0,0 +1,87 @@
+use starknet_in_rust::{
+    core::transaction_hash::{calculate_transaction_hash_common, TransactionHashPrefix},
+    definitions::constants::VALIDATE_DECLARE_ENTRY_POINT_SELECTOR,
+};
+use starknet_types::{
+    contract_address::ContractAddress,
+    contract_class::ContractClass,
+    error::Error,
+    felt::{ClassHash, Felt, TransactionHash},
+    traits::HashProducer,
+};
+
+use crate::constants;
+
+#[derive(Clone, PartialEq, Eq)]
+pub struct DeclareTransactionV1 {
+    pub sender_address: ContractAddress,
+    pub version: Felt,
+    pub max_fee: u128,
+    pub signature: Vec<Felt>,
+    pub nonce: Felt,
+    pub contract_class: ContractClass,
+    pub class_hash: Option<ClassHash>,
```

Comment:
Because of the need to use different implementation than the one provided by starknet_in_rust. Due to the fact that in the version of starknet_in_rust that is used there are some differences in the class hash computation.

Reply:
Ok, so their wrong hash calculation is responsible. Let's hope it's fixed in starknet_in_rust 0.2.0, but for now we can keep it as it currently is. I'll resolve the conversation.